src/verilator/dpi_manager.cc

```diff
@@ -45,11 +45,28 @@ VerilatorDinoCPU * dinoCPU = nullptr;
 //will run a doFetch from within blackbox wrapper
 int ifetch (int imem_address, void* handle){
   DPRINTF(Verilator, "DPI INST FETCH MADE\n");
-
   VerilatorMemBlackBox* hndl = static_cast<VerilatorMemBlackBox *>(handle);
-  hndl->doFetch(imem_address);
+  //lazy exception handeling
+  if (imem_address == 0x80000000 ){
+    DPRINTF(Verilator, "DPI INST FETCH ILLEGAL INST EXCEPTION\n");
+    return 0x00c0006f;//jal x0, 12; effectively halt the processor
+  }else if ( imem_address == 0x80000004 ){
+    DPRINTF(Verilator, "DPI INST FETCH TRAPPING FOR ECALL\n");
+    return 0x0080006f;//jal x0, 8; effectively halt the processor
+  }else if ( imem_address == 0x80000008 ){
+    DPRINTF(Verilator, "DPI INST FETCH TRAPPING FOR EBREAK\n");
+    return 0x0040006f;//jal x0, 4; effectively halt the processor
+  }else if ( imem_address > 0x80000008 ){
```

Comment:
Can your instructions never be mapped beyond this address?

Reply:
they easily can but the processor in question is 32 bit so I was intending for the remaining bytes to be reserved for trap/exception handlers. I currently don't redirect the cpu to such handlers because I don't have any that I've written or available to me so I just stall the processor if a trap or exception is raised. Any ideas on how to handle this better?